File: app/src/main/java/knf/kuma/tv/anime/AnimePresenter.java
package knf.kuma.tv.anime;

import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;
import knf.kuma.pojos.AnimeObject;
import knf.kuma.tv.cards.AnimeCardView;

public class AnimePresenter extends Presenter {

    public AnimePresenter() {
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        return new ViewHolder(new AnimeCardView(parent.getContext()));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, Object item) {
        ((AnimeCardView) viewHolder.view).bind((AnimeObject) item);
    }

    @Override
    public void onUnbindViewHolder(ViewHolder viewHolder) {

    }
}

File: app/src/main/java/knf/kuma/tv/anime/AnimePresenter.kt
package knf.kuma.tv.anime

import android.view.ViewGroup
import androidx.leanback.widget.Presenter
import knf.kuma.pojos.av1.DirectoryAV1Min
import knf.kuma.tv.cards.AnimeCardView

class AnimePresenter : Presenter() {

    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        return ViewHolder(AnimeCardView(parent.context))
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, item: Any?) {
        if (item == null) return
        (viewHolder.view as AnimeCardView).bind(item as DirectoryAV1Min)
    }

    override fun onUnbindViewHolder(viewHolder: ViewHolder) {

    }
}
